core/unit_test/TestTeamCombinedReducers.hpp

```diff
@@ -0,0 +1,459 @@
+
+    Kokkos::parallel_for(
+        policy, KOKKOS_LAMBDA(team_member_type const& team) {
+          auto teamVectorRange = Kokkos::TeamVectorRange(team, n);
+          int teamResult0, teamResult1, teamResult2, teamResult3;
+
+          Kokkos::parallel_reduce(
+              teamVectorRange,
+              [=](int const& i, int& localVal0, int& localVal1, int& localVal2,
+                  int& localVal3) {
+                localVal0 += i + 1;
+                localVal1 *= n;
+                localVal2 = (localVal2 > (i + 1)) ? (i + 1) : localVal2;
+                localVal3 = (localVal3 < (i + 1)) ? (i + 1) : localVal3;
+              },
+              Kokkos::Sum<int>(teamResult0), Kokkos::Prod<int>(teamResult1),
+              Kokkos::Min<int>(teamResult2), Kokkos::Max<int>(teamResult3));
+
+          Kokkos::single(Kokkos::PerTeam(team), [=]() {
+            teamView(0) = teamResult0;
+            teamView(1) = teamResult1;
+            teamView(2) = teamResult2;
+            teamView(3) = teamResult3;
+          });
+        });
+
+    auto hostView = Kokkos::create_mirror_view_and_copy(
+        Kokkos::DefaultHostExecutionSpace(), teamView);
+
+    EXPECT_EQ((n * (n + 1) / 2), hostView(0));
+    EXPECT_EQ((n == 0) ? 0 : std::pow(n, n), hostView(1));
+    EXPECT_EQ((n == 0) ? 0 : 1, hostView(2));
+    EXPECT_EQ(n, hostView(3));
+  }
+
+  void test_team_vector_range_combined_reducers(const int n) {
+    auto policy = Kokkos::TeamPolicy<TEST_EXECSPACE>(1, Kokkos::AUTO);
+    using team_member_type = decltype(policy)::member_type;
+
+    auto teamView = Kokkos::View<int[4], TEST_EXECSPACE::memory_space>("view");
+
+    Kokkos::parallel_for(
+        policy, KOKKOS_LAMBDA(team_member_type const& team) {
+          auto teamVectorRange = Kokkos::TeamVectorRange(team, n);
+          int teamResult0, teamResult1, teamResult2, teamResult3;
+
+          Kokkos::parallel_reduce(
+              teamVectorRange,
+              [=](int const& i, int& localVal0, int& localVal1, int& localVal2,
+                  int& localVal3) {
+                localVal0 += i + 1;
+                localVal1 += i + 1;
+                localVal2 = (localVal2 < (i + 1)) ? (i + 1) : localVal2;
+                localVal3 += n;
+              },
+              teamResult0, Kokkos::Sum<int>(teamResult1),
+              Kokkos::Max<int>(teamResult2), teamResult3);
+
+          Kokkos::single(Kokkos::PerTeam(team), [=]() {
+            teamView(0) = teamResult0;
+            teamView(1) = teamResult1;
+            teamView(2) = teamResult2;
+            teamView(3) = teamResult3;
+          });
+        });
+
+    auto hostView = Kokkos::create_mirror_view_and_copy(
+        Kokkos::DefaultHostExecutionSpace(), teamView);
+
+    EXPECT_EQ((n * (n + 1) / 2), hostView(0));
+    EXPECT_EQ((n * (n + 1) / 2), hostView(1));
+    EXPECT_EQ((n == 0) ? Kokkos::reduction_identity<int>::max() : n,
+              hostView(2));
+    EXPECT_EQ(n * n, hostView(3));
```

Comment:
Technically they all should be reduction_identity if n is equal to 0
```suggestion
    if (n == 0) {
      EXPECT_EQ(Kokkos::reduction_identity<int>::sum(), hostView(0));
      EXPECT_EQ(Kokkos::reduction_identity<int>::sum(), hostView(1));
      EXPECT_EQ(Kokkos::reduction_identity<int>::max(), hostView(2));
      EXPECT_EQ(Kokkos::reduction_identity<int>::sum(), hostView(3));
    } else {
      EXPECT_EQ((n * (n + 1) / 2), hostView(0));
      EXPECT_EQ((n * (n + 1) / 2), hostView(1));
      EXPECT_EQ(n, hostView(2));
      EXPECT_EQ(n * n, hostView(3));
    }
```
Same comment everywhere.